client/app/components/OnboardingTopBar/OnboardingTopBar.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { PropTypes } from 'react';
+
+import css from './OnboardingTopBar.scss';
+
+class OnboardingTopBar extends React.Component {
+
+  t(translationKey) {
+    return this.props.translations[translationKey];
+  }
+
+  next() {
+    if (!this.props.status.slogan_and_description) {
+      return {
+        title: this.t('add_slogan'),
+        link: this.props.guide_root + '/slogan_and_description',
+      };
+    } else if (!this.props.status.cover_photo) {
+      return {
+        title: this.t('add_cover'),
+        link: this.props.guide_root + '/cover_photo',
+      };
+    } else if (!this.props.status.filter) {
+      return {
+        title: this.t('add_filters'),
+        link: this.props.guide_root + '/filter',
+      };
+    } else if (!this.props.status.paypal) {
+      return {
+        title: this.t('add_payments'),
+        link: this.props.guide_root + '/paypal',
+      };
+    } else if (!this.props.status.listing) {
+      return {
+        title: this.t('add_listing'),
+        link: this.props.guide_root + '/listing',
+      };
+    } else if (!this.props.status.invitation) {
+      return {
+        title: this.t('invite_users'),
+        link: this.props.guide_root + '/invitation',
+      };
+    }
+    return null;
+  }
+
+  progress() {
+    const steps = Object.keys(this.props.status).filter((entry) =>
+      ['slogan_and_description',
+       'cover_photo',
+       'filter',
+       'paypal',
+       'listing',
+       'invitation'].includes(entry)).map((key) => [key, this.props.status[key]]);
+    const completed = steps.filter((entry) => entry[1]);
+    return 100 * completed.length / steps.length;
+  }
+
+  nextElement() {
+    if (this.next()) {
+      return (
+        <div className={css.nextContainer}>
+          <div className={css.nextLabel}>
+            {this.t('next_step')}:
+          </div>
+          <a href={this.next().link} className={css.nextButton}>
+            <span>{this.next().title}</span>
+          </a>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  render() {
+    return (
+      <div className={css.topbarContainer}>
+        <div className={css.topbar}>
+          <div className={css.progressLabel}>
+            {this.t('progress_label')}:
+            <span className={css.progressLabelPercentage}>{this.progress().toPrecision(2)} %</span>
+          </div>
+          <div className={css.progressBarBackground}>
+            <div className={css.progressBar} style={{ width: `${this.progress()}%` }} />
+          </div>
+          {this.nextElement()}
+        </div>
+      </div>
+    );
+  }
+}
+
+OnboardingTopBar.propTypes = {
+  translations: PropTypes.object,
+  status: PropTypes.object,
```

Comment:
Using the `PropTypes.object` as a type check isn't too valuable IMO. I'd go as precise as possible, preferably with the `isRequired` validation.

Examples:

```javascript
propTypes = {
  // data1: {"key1": true, "key2": false}
  data1: PropTypes.objectOf(PropTypes.bool).isRequired,
  // data2: [{"username": "mikko", "isAdmin": false}, {"username": "maija", idAdmin: true}]
  data2: PropTypes.arrayOf(PropTypes.shape({
    username: PropTypes.string.isRequired,
    isAdmin: PropTypes.bool.isRequired
  })).isRequired
}
```

As you can see, while quite verbose, you can do very detailed type checking with the PropTypes. I think we would benefit from it in the long run.